### PR TITLE
[DISCUSSION] Load default configuration from global dirac.cfg as well

### PR DESCRIPTION
**Pull request by chaen**
This is not tested bullet proof, but toying with it works. 

This is what I had in mind this morning.
If we put the `BEGIN COMPONENTS` and `END` markers in the dirac.cfg:
* we have a very single place where every possible options for all the sections of the CS will be defined 
* we can extract it from here to build the docs of the agents (`literalinclude`)
* We do not need the ConfigTemplate anymore, although if it is there, the behavior will not change, the two would just be merged, but I would just remove them. 

The main interest I see by doing this is that the information is NEVER repeated.

Now as I said, not much tested, just on a toy like that

```
### BEGIN AgentBefore
AgentBefore
{
  blabla
}
### END
### BEGIN AGENT
Agent
{
  opt1 = toto
  Subsection
  {
    opt2 = toto
  }
}
### END
### BEGIN AgentAfter
AgentAfter
{
  blabla
}
### END
```

@andresailer: 1 line bash
```
[chaen@pclhcb31 tmp]$ sed -n '/### BEGIN AGENT/,/### END/p' dirac.cfg 
### BEGIN AGENT
Agent
{
  opt1 = toto
  Subsection
  {
    opt2 = toto
  }
}
### END
```

### CODE REVIEW

File: FrameworkSystem/Client/ComponentInstaller.py [hunk 1]
```diff
@@ -810,6 +810,28 @@ def getComponentCfg(self, componentType, system, component, compInstance, extens
...
 
+        # Path to the root dirac.cfg
+        cfgTemplatePath = os.path.join(rootPath, 'dirac.cfg')
+        # Read it and extract the interesting part
+        if os.path.exists(cfgTemplatePath):
+          gLogger.notice('Extracting default configuration from', cfgTemplatePath)
+          with open(cfgTemplatePath, 'r') as diracCfgFile:
+            diracCfgContent = diracCfgFile.read()
+            # We take all what is between the BEGIN and END marker of this component
+            componentRegex = r'### BEGIN %s\n(.*?)### END'%componentModule
```
Comment (andresailer) [inline FrameworkSystem/Client/ComponentInstaller.py]: This gives, e.g.
```
   TransformationCleaningAgent
   {...}
```
But what is needed is, e.g.
```
Agents
{
   TransformationCleaningAgent
   {...}
}
```
Why not just load all of dirac.cfg and look for Systems/<System>/[Agents|Services|Executors]/componentModule

**Comment by andresailer** [comment]
While we could remove the ConfigTemplates from DIRAC, the functionality is needed for installing components from extensions as well.

And clearly these changes  are more than one line of bash ;)

**Comment by chaen** [comment]
Oh right. Yeah, maybe loading the full dirac.cfg is an option.
but the more I think of it, the less I think it is fisable to use dirac.cfg as default. The aim I had in mind for this file was to list all possible options. So it makes it an very good input for documentation I would say.
But it is not good for default. 
Now, maybe this shifts a little bit the discussion. Do we want:
* an almost empty CS, with only the options different from the defaults 
* a CS full with every possible options. 

I think I like the 1st option better, providing that there is documentation... 
Ntoe that at the moment, the defaults are repeated twice: in the code, and in the config template

**Comment by fstagni** [comment]
So, to come back on this: it seems to me that we are going in the direction of using the `ConfigTemplate.cfg` as THE source of info for the `/Systems` section, and this can be _..literalInclude(d)_ in the agents + services docstrings in the .py files, and possibly also in the dirac.cfg file.
This means of course making sure that what is the advertised default in the ConfigTemplate.cfg is in sync with the default in the code.

**Comment by chaen** [comment]
You can't do `literalinclude` in the dirac.cvg file 
But I would like to repeat my question: do we really need defaults in the configuration ?

**Pull request closed without merging**